rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -128,13 +145,16 @@ class PublisherBase
   operator==(const rmw_gid_t * gid) const;
 
   using StoreMessageCallbackT = std::function<uint64_t(uint64_t, void *, const std::type_info &)>;
+  using IntraProcessManagerSharedPtr =
+    std::shared_ptr<rclcpp::intra_process_manager::IntraProcessManager>;
```

Comment:
@ivanpauno I think this one is defined for you by [`RCLCPP_SMART_PTR_DEFINITIONS(IntraProcessManager)`](https://github.com/ros2/rclcpp/pull/628/files#diff-c809d7ba52f5bd9fe3dbadb55e8e16feR128) already.

Reply:
I forward declared the class (if not I have some circular including), so I can't access to it.

Reply:
Oops, that's true. May I ask you to add a note near the forward declaration stating why we're doing so?